pkg/render/kubecontrollers/kube-controllers.go

```diff
@@ -99,36 +113,54 @@ func (c *kubeControllersComponent) SupportedOSType() rmeta.OSType {
 }
 
 func (c *kubeControllersComponent) Objects() ([]client.Object, []client.Object) {
+	var kubeControllerServiceAccountName, roleName, roleBindingName, kubeControllerName string
+	if c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise && c.cfg.LogStorageExists &&
```

Comment:
This seems like an overly complicated way of signalling which kube-controller this is. I think that if this is going to be a single render component, then you should probably have all this figured out in the constructor so you don't a barrage of if statements riddled through out this component (when I'd mentioned doing it this way I said it only made sense to do it this way if we didn't have to have a bunch of if statements everywhere to decide which controller this is being rendered for).

What I would try to do is see if it makes sense to have two constructors for this component, and store the configuration in the component. One constructore would be NewElasticsearchKubeControllers, and the other NewSomethingElseKubecontrollers. Then you can set the controller specific information in the object and not have to figure out at different points in this code which controller this is.

DM me if you need some clarification on this.